Fabulous.XamarinForms/src/Fabulous.XamarinForms/Xamarin.Forms.Core.json

```diff
@@ -1,2837 +1,2408 @@
-        },
-        {
-          "name": "HorizontalScrollBarVisibility",
-          "defaultValue": "Xamarin.Forms.ScrollBarVisibility.Default"
-        },
-        {
-          "name": "VerticalScrollBarVisibility",
-          "defaultValue": "Xamarin.Forms.ScrollBarVisibility.Default"
-        },
-        {
-          "name": "RefreshControlColor",
-          "defaultValue": "Xamarin.Forms.Color.Default"
-        }
-      ]
-    },
-    {
-      "name": "Xamarin.Forms.ListView",
-      "modelName": "ListViewGrouped",
-      "customType": "Fabulous.XamarinForms.CustomGroupListView",
-      "members": [
-        {
-          "name": "ItemsSource",
-          "uniqueName": "ListViewGrouped_ItemsSource",
-          "shortName": "items",
-          "inputType": "(string * ViewElement * ViewElement list) list",
-          "modelType": "(string * ViewElement * ViewElement[])[]",
-          "convToModel": "(fun es -> es |> Array.ofList |> Array.map (fun (g, e, l) -> (g, e, Array.ofList l)))",
-          "updateCode": "updateListViewGroupedItems"
-        },
-        {
-          "name": "ShowJumpList",
-          "uniqueName": "ListViewGrouped_ShowJumpList",
-          "inputType": "bool",
-          "modelType": "bool",
-          "defaultValue": "false",
-          "updateCode": "updateListViewGroupedShowJumpList"
-        },
-        {
-          "name": "Footer",
-          "defaultValue": "null",
-          "inputType": "obj",
-          "convToModel": "makeViewOrString"
-        },
-        {
-          "name": "HasUnevenRows",
-          "defaultValue": "false"
-        },
-        {
-          "name": "Header",
-          "defaultValue": "null",
-          "inputType": "obj",
-          "convToModel": "makeViewOrString"
-        },
-        {
-          "name": "IsPullToRefreshEnabled",
-          "defaultValue": "false"
-        },
-        {
-          "name": "IsRefreshing",
-          "defaultValue": "false"
-        },
-        {
-          "name": "RefreshCommand",
-          "defaultValue": "null",
-          "inputType": "unit -> unit",
-          "convToModel": "makeCommand"
-        },
-        {
-          "name": "RowHeight",
-          "defaultValue": "-1"
-        },
-        {
-          "name": "SelectedItem",
-          "uniqueName": "ListViewGrouped_SelectedItem",
-          "defaultValue": "null",
-          "modelType": "(int * int) option",
-          "convToValue": "(function None -> null | Some (i,j) -> let items = target.ItemsSource :?> System.Collections.Generic.IList<ListGroupData> in (if i >= 0 && i < items.Count then (let items2 = items.[i] in if j >= 0 && j < items2.Count then items2.[j] else null) else null))"
-        },
-        {
-          "name": "SeparatorVisibility",
-          "defaultValue": "Xamarin.Forms.SeparatorVisibility.Default"
-        },
-        {
-          "name": "SeparatorColor",
-          "defaultValue": "Xamarin.Forms.Color.D
```

Comment:
It's related to the ordering of the members when writing `View.Entry(...)`.

Today, the order of the constructor members is dependent of the order of declaration in the JSON file.

For example, we have declared `Text` first for `Label`.
This allows us to write `View.Label "Hello World"` without specifying `text=`. 

But this JSON-dependent order has several limitations:
- Harder to maintain the JSON because the properties are not declared alphabetically, whereas MSDN shows them alphabetically sorted.
- Not possible to put before a member of an inherited type if the new type also has its own properties. `View.SomeSuperType(propSuperA, propSuperB, propInheritedA, propInheritedB)`

So I made the order of declaration not relevant, thus we can make it alphabetically sorted.

From what I could see among all the Fabulous apps, only the first constructor member seemed to be used implicitly.
So I created added `PrimaryConstructorMember` field. It allows us to declare which property/event (inherited or not) will be the first member when using the control.

Reply:
Great stuff! 
I didn't think about this.